snapcraft/tests/sources/test_bazaar.py

```diff
@@ -14,12 +14,47 @@
 # You should have received a copy of the GNU General Public License
 # along with this program.  If not, see <http://www.gnu.org/licenses/>.
 
+import os
+import shutil
+import subprocess
+from unittest import mock
+
 from snapcraft.internal import sources
+from snapcraft import tests
+
+
+class BazaarBaseTestCase(tests.sources.SourceTestCase):
+
+    def call(self, cmd):
+        """Call a command ignoring output."""
+        subprocess.check_call(
+            cmd, stdout=subprocess.DEVNULL, stderr=subprocess.DEVNULL)
+
+    def call_with_output(self, cmd):
+        """Return command output converted to a string."""
+        return subprocess.check_output(cmd).decode('utf-8').strip()
 
-from snapcraft.tests.sources import SourceTestCase
+    def rm_dir(self, dir):
+        if os.path.exists(dir):
+            shutil.rmtree(dir)
 
+    def clean_dir(self, dir):
+        self.rm_dir(dir)
```

Comment:
why would this dir exists? aren't we running all tests in a clean dir?
mkdir and not taking care about any clean up should work. If not, we need to add the temp dir fixture to the base test case.